src/django_pg_migration_tools/operations.py

```diff
@@ -9,7 +9,7 @@
 from django.db.migrations.operations import models as operation_models
 
 
-class BaseIndexOperation(
+class SafeIndexOperationManager(
     psql_operations.NotInTransactionMixin,
     base_operations.Operation,
 ):
```

Comment:
In a future refactoring I want to make this class stop inheriting from `NotInTransaction` and `Operation`.

We should have a `MigrationChecker` class that is responsible for addressing whether a migration can happen or not.
This will make it easier to share behaviour between safer index operations and safer constraint operations.

Most of the refactoring here will help with this move. In the interests of not having a big refactoring PR this will be addressed in -> https://github.com/kraken-tech/django-pg-migration-tools/issues/40